enterprise/cmd/frontend/internal/authz/resolvers/repositories.go

```diff
@@ -35,39 +35,65 @@ type repositoryConnectionResolver struct {
 // 🚨 SECURITY: It is the caller's responsibility to ensure the current authenticated user
 // is the site admin because this method computes data from all available information in
 // the database.
+// This function takes returns a pagination of the repo IDs
+// r.ids - the full slice of sorted repo IDs
+// r.after - (optional) the repo ID to start the paging after (does not include the after ID itself)
+// r.first - the # of repo IDs to return
 func (r *repositoryConnectionResolver) compute(ctx context.Context) ([]*types.Repo, *graphqlutil.PageInfo, error) {
 	r.once.Do(func() {
-		// Create the bitmap iterator and advance to the next value of r.after.
-		var afterID api.RepoID
-		if r.after != nil {
-			afterID, r.err = graphqlbackend.UnmarshalRepositoryID(graphql.ID(*r.after))
-			if r.err != nil {
+		var idSubset []int32
+		found := false
+		if r.after == nil {
+			idSubset = r.ids
+		} else {
+			afterID, err := graphqlbackend.UnmarshalRepositoryID(graphql.ID(*r.after))
+			if err != nil {
+				r.err = err
 				return
 			}
-		}
-		iter := r.ids.Iterator()
-		iter.AdvanceIfNeeded(uint32(afterID) + 1) // Plus 1 since r.after should be excluded.
-
-		repoIDs := make([]api.RepoID, 0, r.first)
-		for iter.HasNext() {
-			repoIDs = append(repoIDs, api.RepoID(iter.Next()))
-			if len(repoIDs) >= int(r.first) {
-				break
+			for idx, id := range r.ids {
+				if id == int32(afterID) {
+					found = true
```

Comment:
From the client's POV, this could break previously working query due to legitimated repo deletions.

For example, if I am querying to list first 2 repos after ID 10, semantically the API should give me 2 repos as long as their IDs are greater than 10, regardless of whether 10 actually exists, e.g. [11, 12], or [13, 18].

Here is what I suggest:
```suggestion
			for idx, id := range r.ids {
				if id >= int32(afterID) {
					found = true
```

Reply:
Great call on the change in behavior, I addressed the comments you made in my most recent commit, lmk what you think